core/adapters/bigquery.ts

```diff
@@ -19,29 +19,38 @@ export class BigQueryAdapter extends Adapter implements IAdapter {
     tableMetadata: dataform.ITableMetadata
   ): Tasks {
     const tasks = Tasks.create();
-    // Drop views/tables first if they exist.
-    if (tableMetadata && tableMetadata.type != this.baseTableType(table.type)) {
+    const refreshIncrementCondition = this.refreshIncrementCondition(runConfig, tableMetadata);
+
+    this.addPreOps(table, this.dataformCoreVersion, tasks, refreshIncrementCondition);
+
+    if (tableMetadata && tableMetadata.type !== this.baseTableType(table.type)) {
       tasks.add(
         Task.statement(this.dropIfExists(table.target, this.oppositeTableType(table.type)))
       );
     }
-    if (table.type == "incremental") {
-      if (runConfig.fullRefresh || !tableMetadata || tableMetadata.type == "view") {
+
+    if (table.type === "incremental") {
+      if (refreshIncrementCondition) {
         tasks.add(Task.statement(this.createOrReplace(table)));
       } else {
         tasks.add(
           Task.statement(
             this.insertInto(
               table.target,
               tableMetadata.fields.map(f => f.name),
-              this.where(table.incrementalQuery || table.query, table.where)
+              this.dataformCoreVersion > "1.4.8"
```

Comment:
this shouldn't change, it breaks users

Reply:
Added change where if the `table.where` protobuf is set then it will follow that, otherwise it will follow the new `${ when(incremental(), ... ) }` pattern. I think users will actually be able to use both, but I won't write a test for this unless you say as `when_incremental` is being deprecated anyways.